### DISCUSSION TITLE
Get total length of a video

### DISCUSSION THREAD

**Discussion by muhammadmahad-debug** [muxinc/elements#811] [Q&A] [answered]
<img width="615" alt="Screenshot 2023-11-08 at 5 54 34 AM" src="https://github.com/muxinc/elements/assets/62625370/7f460a32-2a4b-4f89-a398-0b57a3250d1c">

<img width="1213" alt="Screenshot 2023-11-08 at 5 54 52 AM" src="https://github.com/muxinc/elements/assets/62625370/72ad72e4-d827-4137-b32a-237a7ea5d4a6">

that is my event how can I get totalLength of a video

**Comment by luwes** [comment]
try something like `document.querySelector('mux-player').duration` in the callback of that event or easier `event.target.duration`

here's the API reference https://github.com/muxinc/elements/blob/main/packages/mux-player/REFERENCE.md#properties

**Reply by muhammadmahad-debug** [reply]
yes but not getting correct duration 
<img width="688" alt="Screenshot 2023-11-09 at 5 26 17 AM" src="https://github.com/muxinc/elements/assets/62625370/31796d5a-d4b0-4481-8f52-23b8a9d8686d">

this is what the totaly duration is 10 min and some sec but when I get duration from e.target.duration it give me 606.3
<img width="679" alt="Screenshot 2023-11-09 at 5 27 26 AM" src="https://github.com/muxinc/elements/assets/62625370/21690772-4623-4a19-9cc0-4fd9ef11736e">

**Reply by cjpillsbury** [reply, marked as answer]
hey @muhammadmahad-debug, unless I'm misunderstanding, 606.3 is the duration in seconds, which would be 10 minutes and 6 seconds (aka 60 * 10 + 6) or 10:06, rounded to the nearest second. Let me know if I'm missing something here.
